src/battle_script_commands.c

```diff
@@ -6200,6 +6200,34 @@ static void Cmd_moveend(void)
                 gBattleStruct->sameMoveTurns[gBattlerAttacker]++;
             gBattleScripting.moveendState++;
             break;
+        case MOVEEND_SET_EVOLUTION_TRACKER:
+            // If the Pokémon needs to keep track of move usage for its evolutions, do it
+            if (originallyUsedMove != MOVE_NONE
+                && GetBattlerSide(gBattlerAttacker) == B_SIDE_PLAYER
+                && !(gBattleTypeFlags & (BATTLE_TYPE_LINK
+                                        | BATTLE_TYPE_EREADER_TRAINER
+                                        | BATTLE_TYPE_RECORDED_LINK
+                                        | BATTLE_TYPE_TRAINER_HILL
+                                        | BATTLE_TYPE_FRONTIER)))
+            {
+                const struct Evolution *evolutions = GetSpeciesEvolutions(gBattleMons[gBattlerAttacker].species);
```

Comment:
Mentioned over on Discord, but `evolutions` could be `NULL` and this will go to hell. I imagine the AI tests that are timing out probably involve species that don't evolve?